sentry-android-core/src/main/java/io/sentry/android/core/SentryLogcatAdapter.java

```diff
@@ -0,0 +1,109 @@
+package io.sentry.android.core;
+
+import android.util.Log;
+import io.sentry.Breadcrumb;
+import io.sentry.Sentry;
+import io.sentry.SentryLevel;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This class replaces {@link android.util.Log} with its own implementations which creates a {@link
+ * io.sentry.Breadcrumb} for each log. It only replaces log functions that meet a minimum level set
+ * by the user on the Sentry Android Gradle Plugin.
+ */
+@ApiStatus.Internal
+public final class SentryLogcatAdapter {
+
+  private static void addAsBreadcrumb(@NotNull String tag, @NotNull SentryLevel level, String msg) {
+    addAsBreadcrumb(tag, level, msg, null);
+  }
+
+  private static void addAsBreadcrumb(
+      @NotNull String tag, @NotNull SentryLevel level, Throwable tr) {
+    addAsBreadcrumb(tag, level, null, tr);
+  }
+
+  private static void addAsBreadcrumb(
+      @NotNull String tag, @NotNull SentryLevel level, String msg, Throwable tr) {
+    Breadcrumb breadcrumb = new Breadcrumb();
+    breadcrumb.setCategory("log");
```

Comment:
```suggestion
    breadcrumb.setCategory("Logcat");
```
would it make sense to have Logcat here? Similar to how we set [Timber](https://github.com/getsentry/sentry-java/blob/d81684e2d1825564da8d0e52454c0e3ede5cd514/sentry-android-timber/src/main/java/io/sentry/android/timber/SentryTimberTree.kt#L290)